security/src/main/java/com/fincity/security/dao/UserDAO.java

```diff
@@ -409,4 +414,114 @@ public Mono<Set<ULong>> getUserListFromClientIds(Set<ULong> clientList) {
 
 		return Mono.just(users);
 	}
+
+	public Mono<Boolean> checkPasswordEqual(ULong userId, String newPassword) {
+
+		return Mono.from(
+
+		        this.dslContext.select(SECURITY_USER.PASSWORD)
+		                .from(SECURITY_USER)
+		                .where(SECURITY_USER.ID.eq(userId))
+		                .limit(1))
+		        .map(Record1::value1)
+		        .map(pass -> pass.equals(newPassword));
+
+	}
+
+	public Mono<Set<String>> getPastPasswords(ULong userId) {
+
+		return Flux.from(
+
+		        this.dslContext.select(SECURITY_PAST_PASSWORDS.PASSWORD)
+		                .from(SECURITY_PAST_PASSWORDS)
+		                .where(SECURITY_PAST_PASSWORDS.USER_ID.eq(userId)))
+		        .map(Record1::value1)
+		        .collectList()
+		        .map(HashSet::new);
+	}
+
+	public Mono<Boolean> updatePassword(ULong userId, String newPassword) {
+
+		return Mono.from(
+
+		        this.dslContext.update(SECURITY_USER)
+		                .set(SECURITY_USER.PASSWORD, newPassword)
+		                .set(SECURITY_USER.NO_FAILED_ATTEMPT, Short.valueOf((short) 0))
+		                .where(SECURITY_USER.ID.eq(userId)))
+		        .map(val -> val > 0);
+	}
+
+	public Mono<Boolean> checkUserActive(ULong userId) {
+		return Mono.from(
+
+		        this.dslContext.select(SECURITY_USER.STATUS_CODE)
+		                .from(SECURITY_USER)
+		                .where(SECURITY_USER.ID.eq(userId)))
+		        .map(Record1::value1)
+		        .filter(status -> status.equals(SecurityUserStatusCode.ACTIVE))
+		        .map(Objects::nonNull);
+	}
+
+	public Mono<Boolean> deleteToken(ULong tokenId) {
+
+		DeleteQuery<SecurityUserTokenRecord> query = this.dslContext.deleteQuery(SECURITY_USER_TOKEN);
+		query.addConditions(SECURITY_USER_TOKEN.ID.eq(tokenId));
+
+		return Mono.from(query)
+		        .map(val -> val > 0);
+
+	}
+
+	public Mono<List<String>> fetchPasswordsById(ULong userId) {
```

Comment:
Rename to fetchPastPasswords. Also, limit the query with the value in the password policy.

Reply:
not using this. Hence removed this dao method.